scripts/controlmodel_ipadapter.py

```diff
@@ -232,6 +346,16 @@ def get_image_embeds(self, clip_vision_output):
         uncond_image_prompt_embeds = self.image_proj_model(torch.zeros_like(clip_image_embeds))
         return image_prompt_embeds, uncond_image_prompt_embeds
 
+    @torch.inference_mode()
+    def get_image_embeds_faceid_plus(self, face_embed, clip_vision_output):
+        face_embed = face_embed['image_embeds'].to(self.device)
+        from annotator.clipvision import clip_vision_h_uc
+        clip_embed = clip_vision_output['hidden_states'][-2].to(device='cpu', dtype=torch.float32)
+        return (
+            self.image_proj_model(face_embed, clip_embed),
+            self.image_proj_model(torch.zeros_like(face_embed), clip_vision_h_uc.to(clip_embed)),
```

Comment:
I am not very sure whether we should also use `clip_vision_vith_uc` for SDXL here.

The ComfyUI implementation just does `torch.zeros_like(clip_embed)` here. Probably need to investigate later on what should be the correct input here.